Source/SuperLinq/Join.FullOuterJoin.cs

```diff
@@ -0,0 +1,273 @@
+﻿namespace SuperLinq;
+
+public static partial class SuperEnumerable
+{
+	/// <summary>
+	/// Performs a full outer join on two heterogeneous sequences,
+	/// returning a <see cref="ValueTuple{T1, T2}"/> containing 
+	/// the elements from <paramref name="left"/> and <paramref name="right"/>,
+	/// if present, and <see langword="null"/> otherwise.
+	/// Additional arguments specify key selection functions.
+	/// </summary>
+	/// <typeparam name="TLeft">The type of elements in the first sequence.</typeparam>
+	/// <typeparam name="TRight">The type of elements in the second sequence.</typeparam>
+	/// <typeparam name="TKey">The type of the key returned by the key selector functions.</typeparam>
+	/// <param name="left">The first sequence.</param>
+	/// <param name="right">The second sequence.</param>
+	/// <param name="joinType">A value indicating which type of join to use.</param>
+	/// <param name="leftKeySelector">Function that projects the key given an element from <paramref name="left"/>.</param>
+	/// <param name="rightKeySelector">Function that projects the key given an element from <paramref name="right"/>.</param>
+	/// <returns>
+	/// A sequence containing results projected from a full
+	/// outer join of the two input sequences.
+	/// </returns>
+	/// <exception cref="ArgumentNullException"><paramref name="left"/> is <see langword="null"/>.</exception>
+	/// <exception cref="ArgumentNullException"><paramref name="right"/> is <see langword="null"/>.</exception>
+	/// <exception cref="ArgumentNullException"><paramref name="leftKeySelector"/> is <see langword="null"/>.</exception>
+	/// <exception cref="ArgumentNullException"><paramref name="rightKeySelector"/> is <see langword="null"/>.</exception>
+	/// <exception cref="ArgumentException"><paramref name="joinType"/> is <see cref="JoinType.Loop"/>.</exception>
+	/// <remarks>
+	/// <para>
+	/// This method uses deferred execution and streams its results.
+	/// </para>
+	/// </remarks>
+	public static IEnumerable<(TLeft? left, TRight? right)> FullOuterJoin<TLeft, TRight, TKey>(
+		this IEnumerable<TLeft> left,
+		IEnumerable<TRight> right,
+		JoinType joinType,
+		Func<TLeft, TKey> leftKeySelector,
+		Func<TRight, TKey> rightKeySelector)
+	{
+		if (joinType == JoinType.Loop)
+			ThrowHelper.ThrowArgumentException(
```

Comment:
Did you consider using `Guard.IsNot<T>`?

Reply:
enums are not supported by the `Guard.Is<>` and `Guard.IsNot<>`, sadly.